sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -494,11 +494,15 @@ class Dataset[T] private[sql](
   /**
    * Prints the plans (logical and physical) to the console for debugging purposes.
    *
+   * Note that temporary views are already resolved when creating `Dataset`. So if
+   * temporary views are changed after that, the output of this command shows the plans
+   * before such changes.
```

Comment:
I think it is clearer to make a note in this doc.

Reply:
I am not sure whether this is the right place for this comment. I mean, I think that putting it here, it is not clear whether the output of the `explain` shows the plan with "old" views, but the dataset is executed with the "new" ones or both use the "old", as it is.

Reply:
This is the main access of explain command for Dataset. If not this place, any other place you will suggest?

Reply:
My point is: this note and this behavior is not specific to `explain`, but it is common to all operations on a dataset. Putting the comment here can be confusing as it may imply that it is not true for other operations on dataset, which is not true. I'd rather put it in the comment of the Dataset class.